bin/lesson_check.py

```diff
@@ -29,19 +29,19 @@
 # specially. This list must include all the Markdown files listed in the
 # 'bin/initialize' script.
 REQUIRED_FILES = {
-    '%/CODE_OF_CONDUCT.md': True,
-    '%/CONTRIBUTING.md': False,
-    '%/LICENSE.md': True,
-    '%/README.md': False,
-    '%/_extras/discuss.md': True,
-    '%/_extras/guide.md': True,
-    '%/index.md': True,
-    '%/reference.md': True,
-    '%/setup.md': True,
+    'CODE_OF_CONDUCT.md': True,
+    'CONTRIBUTING.md': False,
+    'LICENSE.md': True,
+    'README.md': False,
+    os.path.join('_extras', 'discuss.md'): True,
+    os.path.join('_extras', 'guide.md'): True,
+    'index.md': True,
+    'reference.md': True,
+    'setup.md': True,
 }
 
 # Episode filename pattern.
-P_EPISODE_FILENAME = re.compile(r'/_episodes/(\d\d)-[-\w]+.md$')
```

Comment:
How do you handle removal of `/_episodes/` here?

Reply:
The removal of the folder itself? We'd have to check for it explicitly. The variable is called 'filename' so I assumed it would suffice to check for the filename itself, not the directory name.

Reply:
You removed `/_episodes/` from the pattern. How do you handle this removal?

Reply:
In `check_fileset`, I split the path explicitly. I only check the regular expression against the basename of the path (ie, the name of the file itself).

```
    # Check episode files' names.
    seen = []
    for filepath in filenames_present:
        if '_episodes' not in filepath:
            continue

        # split path to check episode name
        fname = os.path.basename(filepath)
        m = P_EPISODE_FILENAME.search(fname)
        if m and m.group(1):
            seen.append(m.group(1))
        else:
            reporter.add(
                None, 'Episode {0} has badly-formatted filename', filepath)
```